From 8e5b6d341d3ea4b2d16c8d43aee20487edf641a9 Mon Sep 17 00:00:00 2001
From: "River, Li" <river.li@intel.com>
Date: Thu, 24 Dec 2015 18:32:17 +0800
Subject: [PATCH 07/10] FOR_UPSTREAM [VPG]: drm/i915: Add trace for retire work
 state.

This commit adds trace to track retire work idle state.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26200
Change-Id: Icd7e2e2b90bac9eb65d1f0132cd97ae262282357
Signed-off-by: River, Li <river.li@intel.com>
Reviewed-on: https://android.intel.com:443/477664
---
 drivers/gpu/drm/i915/i915_gem.c   |  2 +-
 drivers/gpu/drm/i915/i915_trace.h | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 642f36c..b24d0c8 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -3350,7 +3350,7 @@ i915_gem_retire_work_handler(struct work_struct *work)
 		idle = i915_gem_retire_requests(dev);
 		mutex_unlock(&dev->struct_mutex);
 	}
-
+	trace_i915_gem_retire_work_handler(dev, idle);
 	if (!idle)
 		queue_retire_work(dev_priv, round_jiffies_up_relative(HZ));
 }
diff --git a/drivers/gpu/drm/i915/i915_trace.h b/drivers/gpu/drm/i915/i915_trace.h
index efa86d1..e367b5b 100644
--- a/drivers/gpu/drm/i915/i915_trace.h
+++ b/drivers/gpu/drm/i915/i915_trace.h
@@ -1348,6 +1348,25 @@ TRACE_EVENT(i915_gem_request_complete_begin,
 		__entry->ring, __entry->seqno, __entry->last_seqno)
 );
 
+TRACE_EVENT(i915_gem_retire_work_handler,
+	TP_PROTO(struct drm_device *dev, bool idle),
+
+	TP_ARGS(dev, idle),
+
+	TP_STRUCT__entry(
+		__field(struct drm_device *, dev)
+		__field(bool, idle)
+	),
+
+	TP_fast_assign(
+		__entry->dev = dev;
+		__entry->idle = idle;
+	),
+
+	TP_printk("dev=%p,idle=%d",
+		__entry->dev, __entry->idle)
+);
+
 #endif /* _I915_TRACE_H_ */
 
 /* This part must be outside protection */
-- 
1.9.1

